Code/Framework/AzFramework/Tests/QualitySystemComponentTests.cpp

```diff
@@ -0,0 +1,242 @@
+            m_settingsRegistry->Set(projectPathKey, (enginePath / "AutomatedTesting").Native());
+            AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_AddRuntimeFilePaths(*m_settingsRegistry);
+
+            m_console = AZ::Interface<AZ::IConsole>::Get();
+        }
+
+        void TearDown() override
+        {
+            m_application->Stop();
+            m_application.reset();
+
+            AZ::SettingsRegistry::Unregister(m_settingsRegistry.get());
+            m_settingsRegistry.reset();
+        }
+
+        void StartApplicationWithSettings(AZStd::string_view settings)
+        {
+            auto result = m_settingsRegistry->MergeSettings(settings,
+                AZ::SettingsRegistryInterface::Format::JsonMergePatch,
+                "");
+            ASSERT_TRUE(result);
+
+            // When the application starts and activates the QualitySystemComponent
+            // it will register CVARS based on what is in the registry
+            AZ::ComponentApplication::Descriptor desc;
+            AZ::ComponentApplication::StartupParameters startupParameters;
+            startupParameters.m_loadSettingsRegistry = false;
+            startupParameters.m_loadAssetCatalog = false;
+            m_application->Start(desc, startupParameters);
+
+            // Without this, the user settings component would attempt to save on finalize/shutdown. Since the file is
+            // shared across the whole engine, if multiple tests are run in parallel, the saving could cause a crash
+            // in the unit tests.
+            AZ::UserSettingsComponentRequestBus::Broadcast(&AZ::UserSettingsComponentRequests::DisableSaveOnFinalize);
+        }
+
+        AZ::IConsole* m_console;
+        AZStd::unique_ptr<AzFramework::Application> m_application;
+        AZStd::unique_ptr<AZ::SettingsRegistryInterface> m_settingsRegistry;
+    };
+
+    TEST_F(QualitySystemComponentTestFixture, QualitySystem_Registers_Group_CVars)
+    {
+        // when the quality system component registers group cvars
+        StartApplicationWithSettings(R"(
+            {
+                "O3DE": {
+                    "Quality": {
+                        "Groups": {
+                            "q_test": {
+                                "Levels": [ "low", "high" ],
+                                "Default": 1,
+                                "Description": "q_test quality group",
+                                "Settings": {
+                                    "q_test_sub": [0,1]
+                                }
+                            },
+                            "q_test_sub": {
+                                "Levels": [ "low", "high" ],
+                                "Default": 0,
+                                "Description": "q_test_sub quality group",
+                                "Settings": {
+                                    "a_cvar": [123,234]
```

Comment:
Since I don't see the Setting CVars being created in QualityCVarGroup, the `a_cvar` CVar would never be set because it doesn't exist.

What occurs with that setting in this case?
It looks like we should add to the test a case for that.

Also can you also update the QualityCVarGroup description to mention that the Settings console variables or console commands are **not created**  by the class. Only the Quality console variables are.

I was confused at first about there being a Setting CVar in this test `a_cvar` that wasn't created and was wondering whether the system would create the setting CVar.

Then the next test, is modify values of existing Settings CVars, which implies that the system requires the settings CVars to be defined outside of the quality system.

Reply:
> Since I don't see the Setting CVars being created in QualityCVarGroup, the `a_cvar` CVar would never be set because it doesn't exist.
> 
> What occurs with that setting in this case?

Currently we just send the commands to the Console system and let it handle what happens if a CVAR isn't defined - however in this case I'm silencing the Console output to avoid spam from setting a ton of CVARs at once so you won't see the warning.  I'm thinking it'd be a nice debug feature to enable in the future -  perhaps a CVAR to let the user see the output of applying all the settings.

I'll update the comments of QualityCVarGroup to make it more clear that only the group CVAR is created, not the ones in settings.